liquibase-core/src/main/java/liquibase/snapshot/ResultSetCache.java

```diff
@@ -255,17 +255,29 @@ protected List<CachedRow> executeAndExtract(String sql, Database database) throw
             return executeAndExtract(sql, database, false);
         }
 
+        protected List<CachedRow> executeAndExtract(Database database, String sql, Object...parameters) throws DatabaseException, SQLException {
+            return executeAndExtract(database, false, sql, parameters);
+        }
+
         protected List<CachedRow> executeAndExtract(String sql, Database database, boolean informixTrimHint)
                 throws DatabaseException, SQLException {
+            return executeAndExtract(database, informixTrimHint, sql);
+        }
+
+        protected List<CachedRow> executeAndExtract(Database database, boolean informixTrimHint, String sql, Object...parameters)
+                throws DatabaseException, SQLException {
             if (sql == null) {
                 return new ArrayList<>();
             }
-            Statement statement = null;
+            PreparedStatement statement = null;
```

Comment:
This means that Liquibase will be using PreparedStatement instead of Statement for all queries that are executed (not only Db2) , even if we don't bind any parameters .
I compared functional tests with executions from other branches and seems it doesn't impact performance, but maybe it is something to keep in mind and this method is called in a lot of places.

Reply:
Do we need to wait for @nvoxland's response before we proceed with this PR, @filipelautert ?